test/es-module/test-esm-import-flag.mjs

```diff
@@ -0,0 +1,168 @@
+
+const cjsEntry = fixtures.path('es-modules', 'cjs-file.cjs');
+const cjsImport = fixtures.fileURL('es-modules', 'cjs-file.cjs');
+const mjsEntry = fixtures.path('es-modules', 'mjs-file.mjs');
+const mjsImport = fixtures.fileURL('es-modules', 'mjs-file.mjs');
+
+
+describe('import modules using --import', { concurrency: true }, () => {
+  it('should import when using --eval', async () => {
+    const { code, signal, stderr, stdout } = await spawnPromisified(
+      execPath,
+      [
+        '--import', mjsImport,
+        '--eval', 'console.log("log")',
+      ]
+    );
+
+    assert.strictEqual(stderr, '');
+    assert.match(stdout, /^\.mjs file\r?\nlog\r?\n$/);
+    assert.strictEqual(code, 0);
+    assert.strictEqual(signal, null);
+  });
+
+  it('should import when using --eval and --input-type=module', async () => {
+    const { code, signal, stderr, stdout } = await spawnPromisified(
+      execPath,
+      [
+        '--import', mjsImport,
+        '--input-type', 'module',
+        '--eval', 'console.log("log")',
+      ]
+    );
+
+    assert.strictEqual(stderr, '');
+    assert.match(stdout, /^\.mjs file\r?\nlog\r?\n$/);
+    assert.strictEqual(code, 0);
+    assert.strictEqual(signal, null);
+  });
+
+  it('should import when main entrypoint is a cjs file', async () => {
+    const { code, signal, stderr, stdout } = await spawnPromisified(
+      execPath,
+      [
+        '--import', mjsImport,
+        cjsEntry,
+      ]
+    );
+
+    assert.strictEqual(stderr, '');
+    assert.match(stdout, /^\.mjs file\r?\n\.cjs file\r?\n$/);
+    assert.strictEqual(code, 0);
+    assert.strictEqual(signal, null);
+  });
+
+  it('should import mjs when entrypoint is a cjs file', async () => {
+    const { code, signal, stderr, stdout } = await spawnPromisified(
+      execPath,
+      [
+        '--import', mjsImport,
+        cjsEntry,
+      ]
+    );
+
+    assert.strictEqual(stderr, '');
+    assert.match(stdout, /^\.mjs file\r?\n\.cjs file\r?\n$/);
+    assert.strictEqual(code, 0);
+    assert.strictEqual(signal, null);
+  });
+
+  it('should import cjs when entrypoint is a mjs file', async () => {
+    const { code, signal, stderr, stdout } = await spawnPromisified(
+      execPath,
+      [
+        '--import', cjsImport,
+        mjsEntry,
+      ]
+    );
+
+    assert.strictEqual(stderr, '');
+    assert.match(stdout, /^\.cjs file\r?\n\.mjs file\r?\n$/);
+    assert.strictEqual(code, 0);
+    assert.strictEqual(signal, null);
+  });
+
+  it('should import mjs when entrypoint is a cjs file', async () => {
+    const { code, signal, stderr, stdout } = await spawnPromisified(
+      execPath,
+      [
+        '--import', mjsImport,
+        cjsEntry,
+      ]
+    );
+
+    assert.strictEqual(stderr, '');
+    assert.match(stdout, /^\.mjs file\r?\n\.cjs file\r?\n$/);
+    assert.strictEqual(code, 0);
+    assert.strictEqual(signal, null);
+  });
+
+  it('should import once a file that is --imported twice', async () => {
```

Comment:
nit: this is sounds a little wonkey

```suggestion
  it('should de-duplicate redundant `--import`s', async () => {
```

(I think that's also more correct: if something were `--import`ed thrice, that should also get de-duplicated, rather than just stopping at the first redundancy.)